workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/Model/Details/UnityComponentDetails.cs

```diff
@@ -51,5 +52,33 @@ public void PopulateFields(DetailsStore store)
                     .AsReadOnly();
             }
         }
+
+        public bool IsValid()
+        {
+            var isValid = true;
+            var componentName = ComponentName;
+
+            var clashingCommands = CommandDetails
+                .Where(commandDetail => commandDetail.CommandName.Equals(componentName));
+            foreach (var clashingCommand in clashingCommands)
+            {
+                isValid = false;
+                Console.Error.WriteLine(
+                    $"Error in component \"{componentName}\". " +
+                    $"Command \"{clashingCommand.RawCommandName}\" clashes with component name.");
+            }
+
+            var clashingEvents = EventDetails
+                .Where(eventDetail => eventDetail.EventName.Equals(componentName));
+            foreach (var clashingEvent in clashingEvents)
+            {
+                isValid = false;
+                Console.Error.WriteLine(
+                    $"Error in component \"{componentName}\". " +
+                    $"Event \"{clashingEvent.RawEventName}\" clashes with component name.");
+            }
+
+            return isValid;
```

Comment:
This could be something like `clashingEvents.Any() || clashingCommands.Any()` and we can drop tracking `isValid` through the function? 

Also helps understand what this condition is imo

Reply:
(Obviously be careful with double iteration over the IEnumerables)